api/projects/views.py

```diff
@@ -50,27 +54,44 @@
                 "ID of the organisation to filter by.",
                 required=False,
                 type=openapi.TYPE_INTEGER,
-            )
+            ),
+            openapi.Parameter(
+                "uuid",
+                openapi.IN_QUERY,
+                "uuid of the project to filter by.",
+                required=False,
+                type=openapi.TYPE_STRING,
+            ),
         ]
     ),
 )
 class ProjectViewSet(viewsets.ModelViewSet):
     serializer_class = ProjectSerializer
-    permission_classes = [IsAuthenticated, ProjectPermissions]
+    permission_classes = [ProjectPermissions | MasterAPIKeyProjectPermissions]
     pagination_class = None
 
     def get_queryset(self):
         user = self.request.user
-        queryset = user.get_permitted_projects(permissions=["VIEW_PROJECT"])
+        if user.is_anonymous:
```

Comment:
This is a _slightly_ odd way to split the logic - would it make more sense to do something like `if request.master_api_key` ?